### ISSUE TITLE
[Feature][Chunked Prefill]: Enable cuda graph for chunked prefill. 

### ISSUE THREAD

**Issue by rkooo567** [vllm-project/vllm#4056]
### 🚀 The feature, motivation and pitch

vLLM only enables cuda graph for decoding-only batches (mainly because it didn't see big perf improvement if batched token length > 256). This behavior is preserved even after chunked prefill is enabled, which means when prefill and decodes are batched together (mixed batch) cuda graph is not enabled. 

Since chunked prefill keeps the max number of batched tokens small (for example, 768), we can ideally turn on cuda graph for mixed batches. 

Before continuing any efforts, we should make sure this will actually improve the performance. I am planning to do this from our internal repo which already enables cuda graph for prefill. 

### Alternatives

Not doing it. If the performance improvement is not good enough, we will keep the current behavior rather than adding complexity.

**Comment by foreverpiano** [comment]
@rkooo567  Does anyone do that now? If not, I can give it a try

**Comment by rkooo567** [comment]
Nope. I am measuring the exact perf benefits, but may not have bandwidth to get it done soon. Besides, can you briefly talk about what's the approach you are thinking? The current attention logic has lots of if/else which is not very suitable for cuda graph. 

There are 2 known solutions here; 

1. Cache it based on (num_prefill_tooks, num_decode_toks, num_prefills). This requires to change caching dynamic (currently it is static in the master). We may need to limit the max number of prefills for each step to reduce the cache size as well. 
2. We cache based on num_tokens, and instead we make attention dead simple (put a batch of prefill/decode to a single attention kernel). This had limitation where the parallelism of many attention libraries are decided outside kernel, meaning it is not working without changing them (see https://github.com/flashinfer-ai/flashinfer/issues/187)

**Comment by fengyuanyu1** [comment]
Hello @rkooo567 , has this feature been implemented?
